signature/policy_eval_signedby_test.go

```diff
@@ -130,26 +137,30 @@ func TestPRSignedByIsSignatureAuthorAccepted(t *testing.T) {
 	require.NoError(t, err)
 	sig, err := ioutil.ReadFile("fixtures/unknown-key.signature")
 	require.NoError(t, err)
-	// Pass a nil pointer to, kind of, test that the return value does not depend on the image parmater..
-	sar, parsedSig, err = pr.isSignatureAuthorAccepted(nil, sig)
-	assertSARRejected(t, sar, parsedSig, err)
+	// Pass a nil pointer to, kind of, test that the return value does not depend on the
+	// image parameter..
+	sar, parsedSig, err = pr.isSignatureAuthorAccepted(m, nil, sig)
+	// FIXME: The error is: errors.InvalidArgumentError("no armored data found") but for
```

Comment:
Is it really `no armored data found` here? I see `InvalidSignatureError{msg:"Invalid GPG signature: (*packet.Signature)(nil)"} `.

And that, in turn, happens because the public key is unknown, i.e. `MessageDetails.SignedByKeyId` is set but `MessageDetails.SignedBy` is `nil` because we don’t have the key; and when `SignedBy` is `nil`, `openpgp.readSignedMessage` does not even set up a `signatureCheckReader`, so `Signature` is never set to non-`nil`.

And, that ultimately gets us an `InvalidSignatureError`, not a `PolicyRequirementError`—but then note that this test case does not require a `PolicyRequirementError`. So that seems to be perfectly fine as far as this test is concerned at least; though the `Invalid GPG signature: (*packet.Signature)(nil)` error message could be improved, as could the `gpgme` handling at this level.